packages/store/src/awareness.ts

```diff
@@ -18,12 +18,28 @@ interface UserInfo {
   color: string;
 }
 
+type Request<
+  Flags extends Record<string, unknown> = BlockSuiteFlags,
+  Key extends keyof Flags = keyof Flags
+> = {
+  id: string;
+  clientId: number;
+  field: Key;
+  value: Flags[Key];
+};
+
+type Response = {
+  id: string;
+};
+
 interface AwarenessState<
   Flags extends Record<string, unknown> = BlockSuiteFlags
 > {
   cursor?: SelectionRange;
   user?: UserInfo;
   flags: Flags;
+  changeRequest?: Request<Flags>[];
```

Comment:
Where are `changeRequest` and `changeResponse` being used? Didn't see them being used in this PR.

Reply:
https://github.com/toeverything/blocksuite/pull/695/files#diff-1f168a5b1666540a4a8099cdf382e91e2590e563317a62e99fcb93b88591adafR198-R205

Reply:
https://github.com/toeverything/blocksuite/pull/695/files#diff-1f168a5b1666540a4a8099cdf382e91e2590e563317a62e99fcb93b88591adafR247-R253

Reply:
Let me clarify the whole process:
1. One use set a remote flag, then we push a request item into `state.changeRequest`
2. Others get `changeRequest` and check if he needs to update the flag
3. If the request accepted, push a `changeResponse`
4. cleanup the duplicated `changeResponse` and `changeRequest` which has same id

Reply:
Well but the actual awareness state is using `request` and `response` as field names, rather than `changeRequest` and `changeResponse`.

Reply:
my bad